### ISSUE TITLE
QueryRules not compatible with OpenSearch

### ISSUE THREAD

**Issue by wirtsi** [searchkit/searchkit#1387]
## The problem

I have defined a Query Rule condition like this

` query_rules: [
      {
        id: "1",
        conditions: [
          [
            {
              context: "context",
              value: ["foo"],
            },
          ],
        ],
        actions: [
          {
            action: "QueryFilter",
            query: "status:FOO",
          },
        ],
      },
`
Which causes this query to be fired against _msearch (when I select the foo context via Configure)

`{"query":{"bool":{"filter":[{"query_string":{"query":"status:FOO"}}],"must":{"function_score":{"query":{"pinned":{"ids":[],"organic":{"match_all":{}}}},"functions":[]}}}},"size":15,"from":0,"_source":{"includes":["foo.supplier","bar.name","product_id"]},"highlight":{"pre_tags":["<em>"],"post_tags":["</em>"],"fields":{"product_id":{"number_of_fragments":0},"review_text":{"number_of_fragments":5,"fragment_size":100},"review_text_original":{"number_of_fragments":5,"fragment_size":100}}},"track_total_hits":true}`

Now it turns out that OpenSearch does not support the pinned keyword yet ... but as I am not using the any PinnedAction, this should not break

My server says
`ES said: b'{"error":{"root_cause":[{"type":"parsing_exception","reason":"unknown query [pinned]","line":1,"col":116}],"type":"x_
content_parse_exception","reason":"[1:116] [bool] failed to parse field [must]","caused_by":{"type":"parsing_exception","reason":"unknown query [pinned]","line":1,"col":116,"caused_by":{"typ
e":"named_object_not_found_exception","reason":"[1:116] unknown field [pinned]"}}},"status":400}`

## Environment

* Searchkit version 4.12.0
* Node.js version: 22.9.0
* Typescript version: 5.2.2
* Next.js version: -
* ElasticSearch/OpenSearch version: OpenSearch 2.17.1

**Comment by joemcelroy** [comment]
Yep surprised OpenSearch doesn't support pinned queries. See issue https://github.com/opensearch-project/OpenSearch/issues/72

If queryrules are used and part of the query, we wrap the query and includes an empty pinned query so query rule actions can populate it. 

https://github.com/searchkit/searchkit/blob/main/packages/searchkit/src/transformRequest.ts#L142-L153

An optimization could be to not include an empty pinned query and test to see if the actions has pinned result. https://www.searchkit.co/docs/query-rules/actions/pin-result

Does mean that opensearch will never support pinned result actions unfortunately.
